librz/asm/arch/avr/avr_disasm.h

```diff
@@ -91,8 +65,20 @@ struct _disassembledInstruction {
 };
 typedef struct _disassembledInstruction disassembledInstruction;
 
+/* The disassembler/decoder instruction context */
+struct _avrDisassembleContext {
+	/* Variable to keep track of long instructions that have been found and are to be printed. */
+	int status;
+	/* Variable to hold the address of the long instructions */
+	uint32_t longAddress;
+	/* A copy of the AVR long instruction, we need to keep this so we know information about the
+	 * instruction (mnemonic, operands) after we've read the next 16-bits from the program file. */
+	disassembledInstruction longInstruction;
+};
+typedef struct _avrDisassembleContext avrDisassembleContext;
+
 /* Disassembles an assembled instruction, including its operands. */
-static int disassembleInstruction(disassembledInstruction *dInstruction, const assembledInstruction aInstruction);
+int disassembleInstruction(avrDisassembleContext *context, disassembledInstruction *dInstruction, const assembledInstruction aInstruction);
```

Comment:
```suggestion
int disassembleInstruction(avrDisassembleContext *context, disassembledInstruction *dInstruction, const assembledInstruction aInstruction);
```
Better call it `avr_disassemble_instruction` or something like that.

What is the relationship between `librz/asm/arch/avr/avr_disasm.h`, `librz/asm/arch/avr/disasm.h` and `librz/asm/arch/avr/avr_instructionset.c`? Wouldn't it maybe make more sense to put them all in a single header and prefix everything in there with avr, including the structs and making them follow regular rizin naming conventions? So maybe not prefix with `rz_` but at least get rid of lowerCamelCase stuff.

Reply:
the namings and files comes from the original ripped code, which i have not touched.
I have added some headers because the code was a big `#include "file.c"`